lib/spack/spack/relocate.py

```diff
@@ -240,44 +229,67 @@ def needs_text_relocation(filetype):
     """
     Check whether the given filetype is text that may need relocation.
     """
-    return ("text" in filetype)
+    return ("ASCII text" in filetype)
 
 
-def relocate_binary(path_name, old_dir, new_dir):
+def relocate_binary(path_names, old_dir, new_dir):
     """
     Change old_dir to new_dir in RPATHs of elf or mach-o file path_name
```

Comment:
I noticed that filter_file could take a file list so I thought I should do the same here.

Reply:
No I think filter_file handles Unicode. It was something else. I have been trying to figure out how to handle symbolic links and I thought the file command was following links before reporting the type.

Reply:
I'm assuming https://github.com/spack/spack/pull/6140#discussion_r149256948 is a response to https://github.com/spack/spack/pull/6140#discussion_r149256612. So is the issue that a file is identified as a different type when a symlink path to the file is checked? I'd assume that you don't need to transform symbolic links since you would visit the source file "eventually"